mmf/models/transformers/base.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import os
+from typing import Any, Dict, List
+
+import torch
+from omegaconf import OmegaConf
+from torch import Tensor, nn
+from transformers import AutoConfig, AutoModel
+
+from mmf.common.typings import DictConfig
+from mmf.models import BaseModel
+from mmf.utils.configuration import get_mmf_cache_dir
+from mmf.utils.modeling import get_optimizer_parameters_for_bert
+
+
+class BaseTransformer(BaseModel):
+    def __init__(self, config: DictConfig):
```

Comment:
We should document the config object here.

Reply:
Added dataclass for the config